plugins/CoreAdminHome/CustomLogo.php

```diff
@@ -299,9 +402,7 @@ private function uploadImage($uploadFieldName, $targetHeight, $userPath)
         imagecolortransparent($newImage, $backgroundColor);
 
         imagecopyresampled($newImage, $image, 0, 0, 0, 0, $targetWidth, $targetHeight, $width, $height);
-        imagepng($newImage, PIWIK_DOCUMENT_ROOT . '/' . $userPath, 3);
-
-        return true;
+        return imagepng($newImage, $path, 3);
```

Comment:
If the function fails, we would previously still return true, so this passes the bool value back to the caller.